### ISSUE TITLE
iOS 11.3 WebApp shows blank video

### ISSUE THREAD

**Issue by codemoaner** [richtr/NoSleep.js#51]
Tried this on iOS 11.3 WebApp (safari under the hood).  It "works" in so far as it does not sleep but it pops up a blank square on a black background covering the WebApp which I am guessing is the video.  Tried basic things like hidden:true and display:none without success.  If I home and return it goes away and the app is still humming along.  Used same JS on android WebApp and all good so high degree of confidence that implementation is correct.

**Comment by hartleybrody** [comment]
Adding a bump because I'm seeing this on the firefox focus browser on iOS 11.4.1.

As soon as I trigger nosleep, a white square video pops up. I can dismiss it, but very strange UX.

**Comment by codemoaner** [comment]
Thumbs up.

**Comment by Sechgulo** [comment]
Yeah same issue here. Also trying to identify WebView through UA or the like, to not inclde nosleep in those scenarios, has been impossible so far...

**Comment by codemoaner** [comment]
I failed to disclose that I was running a web app on top of a thin native client.  You need to add the following snippet to your view controller in order to prevent it from popping the noSleep.js video to full screen.

                var config = new WKWebViewConfiguration

               // override Fullscreen mode for video and autoplay

                config.Preferences.JavaScriptEnabled = true;
                config.MediaPlaybackRequiresUserAction = false;
                config.AllowsInlineMediaPlayback = true;

**Comment by hartleybrody** [comment]
That might work for those running a native mobile app. 

Any ideas for those running regular ol' javascript websites in a mobile browser?

**Comment by codemoaner** [comment]
Haven't had a problem otherwise (tested on chrome and FF and Safari non-native on PC, Android and iPhone without issue) but the way I discovered the problem was a little experiment where I substituted my own conspicuous video to watch what was happening and when I did, it showed me my video in the requested dimensions and when I went to no sleep it popped up my video to full screen (whether I liked it or not - i.e. no manner of CSS styling affected this).  In this particular case since the video supplied in noSleep.js is "tiny" (perhaps not visible to the naked eye the the full screen pops up a white box but what it really is doing is making the video full screen - unfortunately is it doesn't show you the video controls (which would have been a useful tip-off).

This led me to conclude that it was not nosleep.js per se  but something in the hosting mechanism that was overriding any settings that I applied to my test video.  I would look at the inline settings that are being employed in noSleep.js and see if whatever is hosting has some other settings that override it and force the video to full screen.  Look for some "inline" setting that may be missing.

Hope that helps....
